kernel/src/table_changes/scan.rs

```diff
@@ -0,0 +1,204 @@
+use std::sync::Arc;
+
+use itertools::Itertools;
+use tracing::debug;
+
+use crate::scan::ColumnType;
+use crate::schema::SchemaRef;
+use crate::{DeltaResult, ExpressionRef};
+
+use super::{TableChanges, CDF_FIELDS};
+
+/// The result of building a [`TableChanges`] scan over a table. This can be used to get a change
+/// data feed from the table
+#[derive(Debug)]
+#[allow(unused)]
+pub struct TableChangesScan {
+    table_changes: Arc<TableChanges>,
+    logical_schema: SchemaRef,
+    predicate: Option<ExpressionRef>,
+    all_fields: Vec<ColumnType>,
+    have_partition_cols: bool,
+}
+
+/// Builder to read the `TableChanges` of a table.
+pub struct TableChangesScanBuilder {
+    table_changes: Arc<TableChanges>,
+    schema: Option<SchemaRef>,
+    predicate: Option<ExpressionRef>,
+}
+
+impl TableChangesScanBuilder {
+    /// Create a new [`TableChangesScanBuilder`] instance.
+    pub fn new(table_changes: impl Into<Arc<TableChanges>>) -> Self {
+        Self {
+            table_changes: table_changes.into(),
+            schema: None,
+            predicate: None,
```

Comment:
could we also do this or does the struct need to implement default?
```suggestion
            ..Default::default()
```

Reply:
Yeah, `..Default::default` doesn't mean "apply default value to each param" -- it means "build a default struct and move fields out of it" (can also do e.g. `..self` in a `with_foo` method).

Reply:
ah interesting thanks!

Reply:
Thx for the explanation of`..Default::default()` @scovich :D

Reply:
Since the scan builder can't have a default (since it needs TableChanges), I'll mark this is resolved.